src/gapstate.h

```diff
@@ -100,6 +100,9 @@ typedef struct GAPState {
 
     UInt1 StateSlots[STATE_SLOTS_SIZE];
 
+    /* For libgap-api.c */
+    Int EnterStackCount;
```

Comment:
Strange--that's surprising to me.  I haven't really looked much into how "HPC-GAP" works but I had assumed that if it was using some native TLS implementation (e.g. with pthreads) that it was just storing the entire `GAPState` in some TLS, and so that should just work regardless who created the threads.  That said, we're not testing against HPC-GAP right now or assuming GAP can even work in a multi-threaded context so if it's easier for now I can just make that a global variable in libgap-api.c.

Reply:
If you run GAP inside a thread it did not start, it also won't have set up any TLS for that thread, and it'll read garbage data and hopefully crash